addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+
+    @Override
+    public void initialize() {
+        start();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+
+        if (command instanceof RefreshType) {
+            refreshData();
+        }
+    }
+
+    public Device[] getLastKnownDevices() {
+        Map<String, Device> temp = jsonSerialNumberDeviceMapping;
+        if (temp == null) {
+            return new Device[0];
+        }
+        Device[] devices = new Device[temp.size()];
+        temp.values().toArray(devices);
+        return devices;
+    }
+
+    public void addEchoHandler(@NonNull EchoHandler echoHandler) {
+        synchronized (echoHandlers) {
+            if (!echoHandlers.contains(echoHandler)) {
+                echoHandlers.add(echoHandler);
+            }
+        }
+        Connection temp = connection;
+        if (temp != null) {
+            initializeEchoHandler(echoHandler, temp);
+        }
+    }
+
+    public void addFlashBriefingProfileHandler(@NonNull FlashBriefingProfileHandler flashBriefingProfileHandler) {
+        synchronized (flashBriefingProfileHandlers) {
+            if (!flashBriefingProfileHandlers.contains(flashBriefingProfileHandler)) {
+                flashBriefingProfileHandlers.add(flashBriefingProfileHandler);
+            }
+        }
+        Connection temp = connection;
+        if (temp != null) {
+            if (currentFlashBriefingJson.isEmpty()) {
+                updateFlashBriefingProfiles(temp);
+            }
+
+            flashBriefingProfileHandler.initialize(this, currentFlashBriefingJson);
+        }
+    }
+
+    public void addSmartHomeHandler(@NonNull SmartHomeBaseHandler smartHomeHandler) {
+        synchronized (smartHomeHandlers) {
+            if (!smartHomeHandlers.contains(smartHomeHandler)) {
+                smartHomeHandlers.add(smartHomeHandler);
+            }
+        }
+        Connection temp = connection;
+        if (temp != null) {
+            smartHomeHandler.initialize(temp);
+        }
+    }
+
+    private void initializeEchoHandler(@NonNull EchoHandler echoHandler, @NonNull Connection temp) {
+        intializeChildDevice(temp, echoHandler);
+
+        Device device = findDeviceJson(echoHandler);
+        BluetoothState state = null;
+
+        JsonBluetoothStates states = null;
+        if (temp.getIsLoggedIn()) {
+            states = temp.getBluetoothConnectionStates();
+        }
+
+        if (states != null) {
+            state = states.findStateByDevice(device);
+        }
+        echoHandler.updateState(device, state);
+    }
+
+    private void intializeChildDevice(@NonNull Connection connection, @NonNull EchoHandler child) {
+        Device deviceJson = this.findDeviceJson(child);
+        if (deviceJson != null) {
+            child.intialize(connection, deviceJson);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+
```

Comment:
Remove this newline, please also check:

http://docs.openhab.org/developers/development/bindings.html
http://docs.openhab.org/developers/development/guidelines.html

And it we would be great if you could also review the report of [the Static code analysis](https://docs.openhab.org/developers/development/bindings.html#static-code-analysis), this also saves me some time while reviewing.

Reply:
Ok